pages/glossary/visible-changes-in-content.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Visible changes in content
+key: visible-changes-in-content
+unambiguous: true
+objective: true
+input_aspects:
+  - CSS styling
+  - DOM tree
+---
+
+An [event][] originates _visible changes in the content_ of a [web page][] when all the following are true
+
+- the [event][] [queues a task][queuing] in the [task source][] of an [event loop][] of a [Document][] of the [web page][]; and
+- the [processing][] of that [task][] causes an [update to the rendering][update the rendering] of the [Document][]; and
```

Comment:
What does it mean that a task causes an update? If the task sets a timeout that eventually leads to an update, does it cause that update, or is this only about changes in that occur in the same tick? 

I don't think this is all that testable. How would anyone know this is true? Are you expecting someone to step through the debugger to figure this out?

Reply:
That task is in the queue of the event loop of the document, being managed by the browser. If the task was created by a timeout that was set an hour ago is irrelevant for this definition. The timeout is the event that places the task in the queue. Which means that the hover event was not the one that placed the (rendering change) task on the queue.

The definition evolved from pixels changed in the viewport caused by some event that I can't be certain of (and thus would require an assumption that no other event was causing the change - which was challenged by Kasper), to this version of the definition so that it is testable.

Reply:
As I understand it, the task queue resolves all tasks as soon as possible. If I use a timeout, it doesn't go into the task queue. Depending on the JS engine, there might be a task to check if any timeouts need to be queued up, but the actual timeout handler doesn't go to the task queue. I don't think what you're saying is correct.

More importantly, the event loop and task queue are not included in the DOM tree. This definition is testing thing that aren't part of the input of the rule. That's not allowed in ACT rules. Can you explain to me how someone would go about testing this? In what testing environment does the browser give you a way to inspect the task queue?

Reply:
You're correct. I don't think we can include "Browser rendering engine" (or any other engine) as an input aspect. 

Which means we can't use this. And therefore we're back to the initial problem of identifying what changes result from the hovering event. This was initially was handled with an assumption. But @kasperisager pointed out it was too strong of an assumption too make.

So, it's back to the drawing board... and I can't clearly see a way to (objectively) tie the event to the visible changes on a page, so if you have any suggestions they're welcome